pjmedia/src/pjmedia/converter_libyuv.c

```diff
@@ -76,7 +76,7 @@ typedef enum conv_func_type
     SCALE_PLANAR
 } conv_func_type;
 
-typedef void (*gen_conv_func)();
+typedef void* gen_conv_func;
```

Comment:
this doesn't seem right.
void (*gen_conv_func)(), is a function pointer declaration, which specifies that gen_conv_func is a pointer to a function that takes an unspecified number of arguments and returns nothing (i.e., void). The syntax (*gen_conv_func) is used to dereference the pointer and access the function itself.

void* gen_conv_func, is a declaration of a void pointer, which specifies that gen_conv_func is a pointer to an unspecified type of data. Void pointers can point to any type of data, but they do not have any associated type information, so you cannot directly dereference them.

Reply:
I use gcc12 on linux test, so many warnings as follow:
```
../src/pjmedia/converter_libyuv.c:163:48: warning: cast between incompatible function types from ‘int (*)(const uint8 *, int,  const uint8 *, int,  const uint8 *, int,  uint8 *, int,  int,  int)’ {aka ‘int (*)(const unsigned char *, int,  const unsigned char *, int,  const unsigned char *, int,  unsigned char *, int,  int,  int)’} to ‘void (*)()’ [-Wcast-function-type]
  163 |         GET_PJ_FORMAT(dst),CONV_PLANAR_TO_PACK,(gen_conv_func)&method
      |                                                ^   
../src/pjmedia/converter_libyuv.c:187:6: note: in expansion of macro ‘MAP_CONV_PLANAR_TO_PACK’
  187 |     {MAP_CONV_PLANAR_TO_PACK(I420,UYVY,I420ToUYVY)},
      |      ^~~~~~~~~~~~~~~~~~~~~~~
../src/pjmedia/converter_libyuv.c:165:50: warning: cast between incompatible function types from ‘int (*)(const uint8 *, int,  const uint8 *, int,  const uint8 *, int,  uint8 *, int,  uint8 *, int,  uint8 *, int,  int,  int)’ {aka ‘int (*)(const unsigned char *, int,  const unsigned char *, int,  const unsigned char *, int,  unsigned char *, int,  unsigned char *, int,  unsigned char *, int,  int,  int)’} to ‘void (*)()’ [-Wcast-function-type]
  165 |         GET_PJ_FORMAT(dst),CONV_PLANAR_TO_PLANAR,(gen_conv_func)&method
      |                                                  ^   
../src/pjmedia/converter_libyuv.c:188:6: note: in expansion of macro ‘MAP_CONV_PLANAR_TO_PLANAR’
  188 |     {MAP_CONV_PLANAR_TO_PLANAR(I420,I422,I420ToI422)},
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~
../src/pjmedia/converter_libyuv.c:169:41: warning: cast between incompatible function types from ‘int (*)(const uint8 *, int,  const uint8 *, int,  const uint8 *, int,  int,  int,  uint8 *, int,  uint8 *, int,  uint8 *, int,  int,  int,  enum FilterMode)’ {aka ‘int (*)(const unsigned char *, int,  const unsigned char *, int,  const unsigned char *, int,  int,  int,  unsigned char *, int,  unsigned char *, int,  unsigned char *, int,  int,  int,  enum FilterMode)’} to ‘void (*)()’ [-Wcast-function-type]
  169 |         GET_PJ_FORMAT(fmt),SCALE_PLANAR,(gen_conv_func)&method

```

after change, warnings disappear

But i use clang17 check it,  both not change and change  no the warnings

Reply:
@sauwming You'd better install lastest gcc12 or clang16/17 on linux to test it

Reply:
@sauwming  clang  CFLAGS   add "-Wcast-function-type" can find out the warning

```bash
# This is the sugguest flags that compatible with both gcc and clang
export CFLAGS= \
"-Wextra -Wno-missing-field-initializers -Wno-unused-but-set-variable 
-Wsingle-bit-bitfield-constant-conversion -Wcast-function-type"
```

Reply:
you can simply change it to `typedef int (*gen_conv_func)();` to get rid of the warning